wagtail/admin/templates/wagtailadmin/pages/action_menu/menu.html

```diff
@@ -1,6 +1,11 @@
 {% if show_menu %}
+    {{ default_menu_item }}
     <div class="dropdown-toggle icon icon-arrow-up"></div>
     <ul>
-        {% for item in rendered_menu_items %}{{ item }}{% endfor %}
+        {% for item in rendered_menu_items %}
+            <li>
+                 {{ item }}
```

Comment:
Moving the `<li>` tags out of the template is a breaking change. I agree it's better this way though. @gasman is this worth an upgrade considerations entry?

Reply:
@kaedroho Probably a good idea, yes. We don't document what should be in the template, but we do document the fact that the template is overrideable, which makes this feature "semi-official".

In fact, thinking about this further, this could be a problem for add-on packages that make use of this hook but want to be compatible with both Wagtail versions. Is there any way of achieving that, other than having two templates and using the Wagtail version number to choose between them?

Reply:
Release note added in https://github.com/gasman/wagtail/commit/ad753ad9de9d7c273a28c332e9d483c29fd3a0c3. I thought about ways to make this fully backwards-compatible (e.g. giving the `<li>`-less template a distinct attribute like `button_template` and having the menu logic choose the best available template), but decided that was overkill in the end - the side effect of an extra `<li>` is minimal (an extra 1px border), and it's easy for a third-party library to add a version check if necessary.